contracts/solidity/control/ControlStructures.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+contract TestContract {
+
+    constructor() {}
+
+    function myFunc(uint x) public pure returns (string memory) {
+        require(x != 0, "require failed");
+        return "my func was called";
+    }
+}
+
+
+contract ControlStructures {
+    TestContract public test;
+
+    constructor() {
+        test = new TestContract();
+    }
+
+    function testIfElse(bool condition) external pure returns(bool) {
+        if(condition) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function testWhile(uint256 total) external pure returns(uint256) {
+        uint256 it = 0;
```

Comment:
Nit: let's put a require max check on the total for this and the other while loops for best practice.
Just so we ensure a reasonable max int value.